plugins/doc_fragments/auth.py

```diff
@@ -27,6 +27,7 @@ class ModuleDocFragment(object):
           - jwt
           - cert
           - none
+          - k8s
```

Comment:
I think we should give it a fuller name and call the method `kubernetes` as they do in Vault itself.
This will need a lot of renames but hopefully it's all simple find/replace.